app/src/main/java/io/github/droidkaigi/confsched2018/data/db/FavoriteFirestoreDatabase.kt

```diff
@@ -47,40 +47,39 @@ class FavoriteFirestoreDatabase : FavoriteDatabase {
         Single.error(NotPreparedException())
     } else {
         getCurrentUser().flatMap { currentUser ->
-            return@flatMap Single.create<Boolean>({ e ->
+            return@flatMap Single.create<DocumentSnapshot>({ e ->
```

Comment:
Can we commonize getting the favorite Single object by creating method?

Reply:
Well... a method like `getFavorite(currentUser, sessionId)` ?

Reply:
I think like this private fun getFavoritesDocument(current):Single<DocumentSnapshot>
And, I think it is better to use this method not only `favorite()` but also `setupFavoritesDocument()`
But, This is not required!
We can merge this 👍

Reply:
`favorite()` requires `"users/${currentUser.uid}/favorites/${session.id}"` (Document),
on the other hand,
 `setupFavoritesDocument()` requires `"users/${currentUser.uid}/favorites"` (Collection).

So, it seems to be difficult to commonize them. 🤔

Reply:
Thanks 👍  I understood!